### PR TITLE
only reload component if absolutely neccesaray

### PR DESCRIPTION
**Pull request by conorhastings**
@rovolution does this logic look correct?

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -6,20 +6,24 @@ export default class AsyncBranch extends React.Component {
...
   this.loadComponent(this.props);
  } 
 
  componentWillReceiveProps(nextProps) {
-  if ((nextProps.condition !== this.props.condition) || (nextProps.left !== this.props.left) || (nextProps.right !== this.props.right)) {
-   this.loadComponent(nextProps);
-  }
+  if (
+    nextProps.condition !== this.props.condition || 
+    (nextProps.condition && nextProps.left !== this.props.condition.left) ||
```
Comment (rovolution) [inline src/index.js]: typo? `this.props.condition.left`-> `this.props.left`?

**Review by rovolution — CHANGES_REQUESTED** [review]
@conorhastings yea this logic definitely makes sense. no need to check if the `left` or `right` values are different unless the condition corresponds to rendering that component.

only thing was the typo (see code comment)

File: src/index.js (hunk 1 above)
Comment (conorhastings) [reply src/index.js]: 👍 nice catch

**Comment by conorhastings** [comment]
@rovolution updated, look good?

**Review by rovolution — APPROVED** [review]
@conorhastings looks good 👍

**Comment by rovolution** [comment]
🆒

**Pull request merged**